### PR TITLE
Documentation for adding NFS mount options to storage pool

### PR DESCRIPTION
**Pull request by abh1sar**
Documentation PR for https://github.com/apache/cloudstack/pull/8947

----
📚 Documentation preview 📚: https://cloudstack-documentation--396.org.readthedocs.build/en/396/

### CODE REVIEW

**Comment by abh1sar** [comment]
@sureshanaparti @rohityadavcloud @shwstppr @harikrishna-patnala please review

**Comment by shwstppr** [comment]
@blueorangutan docbuild

File: source/adminguide/storage.rst [hunk 1]
```diff
@@ -237,6 +238,66 @@ templates, and ISOs.
...
+Although it depends on the NFS server, but commonly supported `vers` values 
+are `3` for NFSv3 and minor versions `4.0, 4.1 and 4.2` for NFSv4. 
+`nconnect` values can range from 1 to 16.
+
+Administrator can give the NFS mount options while adding a Primary Storage
+from the Create Zone Wizard as well as the Add Primary Storage form.
+|nfs-mount-options-create-zone-wizard.png|
+|nfs-mount-options-add-primary-storage.png|
+
+NFS mount options can be changed on a pre-existing Storage Pool using the
```
Comment (sureshanaparti) [inline source/adminguide/storage.rst:259]:
```suggestion
NFS mount options can be changed on a pre-existing Storage Pool in maintenance mode, using the
```

File: source/adminguide/storage.rst [hunk 2]
```diff
@@ -237,6 +238,66 @@ templates, and ISOs.
...
+Edit Primary Storage form. Storage Pool needs to be unmounted and mounted
+again on the KVM host using the new options. Running VMs using volumes
```
Comment (sureshanaparti) [inline source/adminguide/storage.rst:261]:
```suggestion
again on the KVM host using the new options upon cancelling the maintenance. Running VMs using volumes
```

File: source/adminguide/storage.rst [hunk 3]
```diff
@@ -237,6 +238,66 @@ templates, and ISOs.
...
+in the Storage Pool either need to be stopped or the volumes migrated.
+So the NFS Mount Options field below is only shown if the Storage Pool
```
Comment (sureshanaparti) [inline source/adminguide/storage.rst:263]: I think, this and below 2 lines may not be required after updated with above suggestions

**Comment by abh1sar** [comment]
@blueorangutan docbuild

File: source/adminguide/storage.rst [hunk 4]
```diff
@@ -237,6 +238,74 @@ templates, and ISOs.
...
+Storage Pool will be unmounted and mounted again on the KVM hosts using the
+new options upon cancelling the maintenance mode.
+|nfs-mount-options-edit-primary-storage.png|
+
+Mount failing due to an incorrect Mount Option
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Add Storage Pool will fail with the error ``An incorrect mount option was specified``.
+
+In the Update storage pool case, cancel maintenance will throw the above error.
+The Administrator has to repeat the process of enabling maintenance mode,
```
Comment (sureshanaparti) [inline source/adminguide/storage.rst:272]: not need to enable maintenance mode again, admin has to change the mount options to the correct one and then cancel the maintenance mode.

**Comment by abh1sar** [comment]
@blueorangutan builddoc

**Comment by sureshanaparti** [comment]
@blueorangutan docbuild

**Review by sureshanaparti — APPROVED** [review]
Doc LGTM

**Review by DaanHoogland — APPROVED** [review]

**Pull request merged**